wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/StatementImpl.java

```diff
@@ -74,6 +74,36 @@
 
 	private List<SnakGroup> qualifiersGroups;
 
+	/**
+	 * Checks if the statement ID consists of a first part
+	 * with the entity id of the item the statement belongs to, the separator $, plus
+	 * a random hash of the form
+	 * /^\{?[A-Z\d]{8}-[A-Z\d]{4}-[A-Z\d]{4}-[A-Z\d]{4}-[A-Z\d]{12}\}?\z/
+	 *
+	 * @param statementID
+	 * @return true if the statement ID is of the valid format
+	 */
+	public static boolean validateStatementID(String statementID)
+	{
+		if(statementID.equals(""))
+			return true;
+
+		String separator = "\\$";
+		String statementFormat ="^\\{?[A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{12}\\}?$";
+
+		String[] statementParts = statementID.split(separator);
+		if(statementParts.length != 2)
+			throw new IllegalArgumentException("Statement ID does not have the correct number of parts");
+
+		if(statementParts[0].charAt(0) != 'Q' && statementParts[0].charAt(0) != 'P' && statementParts[0].charAt(0) != 'L' && statementParts[0].charAt(0) != 'M')
```

Comment:
There are actually more types (forms, senses, media-info...). You could use `EntityIdValueImpl.guessEntityTypeFromId` that will does the same thing and supports all types.